plugins/modules/win_dns_record.ps1

```diff
@@ -150,11 +150,14 @@ if ($null -ne $values -and $values.Count -gt 0) {
         }
         try {
             if ($type -eq 'SRV') {
-                Add-DnsServerResourceRecord -SRV -Name $name  -ZoneName $zone @srv_args @extra_args -WhatIf:$module.CheckMode
+                Add-DnsServerResourceRecord -SRV -Name $name -ZoneName $zone @srv_args @extra_args -WhatIf:$module.CheckMode
             }
             elseif ($type -eq 'TXT') {
                 Add-DnsServerResourceRecord -TXT -Name $name -DescriptiveText $value -ZoneName $zone -TimeToLive $ttl @extra_args -WhatIf:$module.CheckMode
             }
+            elseif ($type -eq 'CNAME') {
+                Add-DnsServerResourceRecordCName -Name $name -HostNameAlias $value -ZoneName $zone -TimeToLive $ttl -WhatIf:$module.CheckMode
```

Comment:
So the changes here seem ok I'm just not sure what to do about the extra arguments like `@extra_args`, `-AllowUpdateAny` and possibly `@splat_args` that are in use in the original call.

`extra_args` contains the `-ComputerName` parameter which `Add-DnsServerResourceRecordCName` does have so it should be safe to add that. I think `splat_args` can probably be ignored as  it contains the CName info that this new cmdlet implicitly knows. I have no idea what `-AllowUpdateAny` update any does but the fact that `Add-DnsServerResourceRecord` also specified it makes me think this new call should.

Reply:
Thanks for your feedback.

- I have added the "@extra_args" to the command. I have also added a comment near the variable declaration to prevent ulterior breaking changes.

- I have not added the "-AllowUpdateAny" for now. This parameter has the same behaviour for both powershell modules so we could add it, but it allows any authenticated users to modify the DNS record which seems as a security problem. Maybe we could add a parameter to the ansible module to allow enabling/disabling this flag. What's your thoughts ?

- I have not added the "@splat_args" based on your comment and the fact that I did not find what this argument does. Is it doing anything at all ? 

- I have added a fragment as requested.